org.eclipse.xtext.common.types/src/org/eclipse/xtext/common/types/descriptions/ClasspathScanner.xtend

```diff
@@ -52,16 +52,22 @@ class ClasspathScanner {
 		return ClassLoader.systemClassLoader.getDescriptors(true, packagePrefixes)
 	}
 	
+	var String[] systemClasspath = null
```

Comment:
Could this variable be accessed by different threads? Is synchronization required?

Reply:
Maybe just a theoretical issue. When 2 threads have null for that field, both would do the same computation and last wins. So if you think that does not matter then I'm fine with that.

Reply:
What bugs me more is that we cache an array which is by definition not immutable. Callers may change the entries.

Reply:
the array is in the signature